### PR TITLE
Use -O3 for release. (Accelerates clang by factor 3.6!)

### PR DESCRIPTION
**Pull request by Axel-Naumann**

### CODE REVIEW

**Comment by dpiparo** [comment]
Hi @Axel-Naumann , great! But here we are building the entire ROOT + cling with O3, something like an OPTIMIZED build - doesn't this potentially introduce changes in the results of the users?
I am not against it in principle, on the contrary, I encourage aggressive flags, but they may have a consequence, right?

File: cmake/modules/SetUpLinux.cmake [hunk 1]
```diff
@@ -122,14 +122,14 @@ if(CMAKE_COMPILER_IS_GNUCXX)
...
+  set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "-O3 -g -DNDEBUG")
+  set(CMAKE_CXX_FLAGS_RELEASE        "-O3 -DNDEBUG")
   set(CMAKE_CXX_FLAGS_OPTIMIZED      "-Ofast -DNDEBUG")
   set(CMAKE_CXX_FLAGS_DEBUG          "-g")
   set(CMAKE_CXX_FLAGS_DEBUGFULL      "-g3")
   set(CMAKE_CXX_FLAGS_PROFILE        "-g3 -ftest-coverage -fprofile-arcs")
-  set(CMAKE_C_FLAGS_RELWITHDEBINFO   "-O2 -g -DNDEBUG")
-  set(CMAKE_C_FLAGS_RELEASE          "-O2 -DNDEBUG")
+  set(CMAKE_C_FLAGS_RELWITHDEBINFO   "-O3 -g -DNDEBUG")
+  set(CMAKE_C_FLAGS_RELEASE          "-O3 -DNDEBUG")
```
Comment (amadio) [inline cmake/modules/SetUpLinux.cmake]: This will change it for the whole of ROOT, is that intended?

Comment (Axel-Naumann) [reply cmake/modules/SetUpLinux.cmake]: Yes that was my intent. If clang benefits from it, we should also allow the experiments and users to benefit from it. Likely esp new code (non-TObject-code) will benefit.

Comment (amadio) [reply cmake/modules/SetUpLinux.cmake]: I wouldn't be so keen on using O3 for the whole of ROOT. The fact that a test failed with this change makes me even more worried.

Comment (dpiparo) [reply cmake/modules/SetUpLinux.cmake]: @axel, but these are the OPTIMIZED builds without fast math - happy to see this here. We have been testing those for a long while, also to be ready to switch to the intel compiler - switch that happened a while ago. I am in favour of this change for 6.14 - this is maybe obvious but I would still keep jitting optimisations down to O0 in a first phase. Maybe now, with the improved cling, we could even try and test more aggressive optimisation levels for jitted code (pinging @yamaguchi1024 , compiler expert)

Comment (dpiparo) [reply cmake/modules/SetUpLinux.cmake]: If we go for this, we should make sure to remove the OPTIMIZED CMAKE_BUILD_TYPE ;-)

Comment (amadio) [reply cmake/modules/SetUpLinux.cmake]: I believe that was already done :-)

Comment (dpiparo) [reply cmake/modules/SetUpLinux.cmake]: oh right, thanks!

Comment (Axel-Naumann) [reply cmake/modules/SetUpLinux.cmake]: That test failure is https://sft.its.cern.ch/jira/browse/ROOT-9174

Given no test failures any possible unintended consequences are speculation - so I'll merge. We will see from the experiments' feedback + LCG nightlies whether actual problems exist.

**Pull request merged**